core/controllers/editor.py

```diff
@@ -780,6 +780,7 @@ class LearnerAnswerInfoHandler(EditorHandler):
     """Handles the learner answer info for an exploration state."""
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    IS_LOGGED_IN_CHECK_REQUIRED = False
```

Comment:
Ah, let's not allow bypassing can_edit_entity decorator via a gust user.

Reply:
Updated the code to have changes in frontend instead of the decorator as per your suggestion.